client/my-sites/domains/domain-search/index.jsx

```diff
@@ -92,13 +92,18 @@ class DomainSearch extends Component {
 	};
 
 	handleAddMapping = ( domain ) => {
-		this.props.shoppingCartManager
-			.addProductsToCart( [
-				fillInSingleCartItemAttributes( domainMapping( { domain } ), this.props.productsList ),
-			] )
-			.then( () => {
-				this.isMounted && page( '/checkout/' + this.props.selectedSiteSlug );
-			} );
+		this.isMounted && page( this.getDomainMappingUrl( domain ) );
+	};
+
+	getDomainMappingUrl = ( domain ) => {
+		let url = '/domains/add/mapping';
+
+		if ( this.props.selectedSiteSlug ) {
+			const query = stringify( { initialQuery: domain.trim() } );
+			url += `/${ this.props.selectedSiteSlug }?${ query }`;
+		}
+
+		return url;
```

Comment:
I think we have a function `domainMapping()` in `client/my-sites/domains/paths.js` that will generate this URL for you.

Reply:
Good call! I'll reuse that function.